pkgs/top-level/haxe-packages.nix

```diff
@@ -52,67 +53,62 @@ let
         '';
 
         meta = {
-          homepage = "http://lib.haxe.org/p/${libname}";
+          homepage = "https://lib.haxe.org/p/${libname}";
           license = lib.licenses.bsd2;
           platforms = lib.platforms.all;
           description = throw "please write meta.description";
         } // attrs.meta;
       });
 
+    inherit haxe;
+
     hxcpp = buildHaxeLib rec {
       libname = "hxcpp";
-      version = "4.1.15";
-      sha256 = "1ybxcvwi4655563fjjgy6xv5c78grjxzadmi3l1ghds48k1rh50p";
+      version = "4.2.1";
+      sha256 = "10ijb8wiflh46bg30gihg7fyxpcf26gibifmq5ylx0fam4r51lhp";
       postFixup = ''
-        for f in $out/lib/haxe/${withCommas libname}/${withCommas version}/{,project/libs/nekoapi/}bin/Linux{,64}/*; do
+        # <xlocale> is not available on both glibc and musl, but it looks that there is no __MUSL__ to detect it
+        substituteInPlace ${placeholder "out"}/lib/haxe/${withCommas libname}/${withCommas version}/src/hx/libs/std/Sys.cpp --replace \
+          '&& !defined(__GLIBC__)' \
+          '&& 0'
+
+        for f in ${placeholder "out"}/lib/haxe/${withCommas libname}/${withCommas version}/{,project/libs/nekoapi/}bin/Linux{,64}/*; do
           chmod +w "$f"
-          patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker)   "$f" || true
-          patchelf --set-rpath ${ lib.makeLibraryPath [ stdenv.cc.cc ] }  "$f" || true
+          patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker)  "$f" || true
+          patchelf --set-rpath       ${ lib.makeLibraryPath [ stdenv.cc.cc ] }  "$f" || true
```

Comment:
```suggestion
          patchelf --set-rpath       ${lib.makeLibraryPath [ stdenv.cc.cc ]}  "$f" || true
```

Reply:
this space has been before.
also it will break matched `"$f" || true` on 2 lines and reduce readability